casper/src/main/scala/coop/rchain/casper/SafetyOracle.scala

```diff
@@ -204,11 +210,19 @@ sealed abstract class SafetyOracleInstances {
         }
       }
 
+      /**
+        * @param blockDag
+        * @param candidate metadata of a block examined
+        * @param target target block hash
+        * @return if candidate block is compatible with target block
+        */
       private def computeCompatibility(
           blockDag: BlockDagRepresentation[F],
-          candidateMetadata: BlockMetadata,
-          targetBlockHash: BlockHash
+          candidate: BlockMetadata,
+          target: BlockHash
       ): F[Boolean] =
-        isInMainChain(blockDag, candidateMetadata, targetBlockHash)
+        // Current implementation checks if candidate block is a parent of target block
+        ProtoUtil.getBlockMetadata[F](target, blockDag) >>=
+          (DagOperations.isDescendantOf[F](_, candidate, blockDag))
```

Comment:
`(DagOperations.isDescendantOf[F](_, candidate, blockDag))` should be `(DagOperations.isDescendantOf[F](candidate, _, blockDag))`

Reply:
target is `latestMessage` of a validator, we should check if the candidate block(the block we want to finalized) isDescendantOf target latestMessage.

Reply:
I believe no, candidate here is a candidate for finalization, so its an ancestor. 

First param in isDescendantOf is a a block being checked if it is a descendant, so the second one our candidate for finalization .

Reply:
Ok. You are right. I am wrong.